dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,553 @@
+        }
+    }
+
+    protected void doAuthenticationLoginSecurityLog(final HttpServletRequest request,
+                                                    final IdentityProviderConfiguration identityProviderConfiguration,
+                                                    final User user) {
+        try {
+
+            final Host   host  = this.hostWebAPI.getCurrentHost(request);
+            final String env   = this.samlWebUtils.isFrontEndLoginPage(request.getRequestURI()) ? "frontend" : "backend";
+            final String log   = new Date() + ": Successfull SAML login for Site '" + host.getHostname() + "' with IdP " +
+                    "ID: " + identityProviderConfiguration.getId() + " (" + env + ") from " + request.getRemoteAddr() + " for user: " +
+                    user.getEmailAddress();
+
+            // “$TIMEDATE: SAML login success for $host (frontend|backend) from $REQUEST_ADDR for user $username”
+            SecurityLogger.logInfo(SecurityLogger.class, this.getClass() + " - " + log);
+            Logger.info(this, log);
+        } catch (Exception e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Perform the logic to get or create the user from the SAML and DotCMS If
+     * the SAML_ART_PARAM_KEY, will resolve the Assertion by calling a Resolver
+     * and will create/get/update the user on the dotcms data.
+     *
+     * @param request {@link HttpServletRequest}
+     * @param identityProviderConfiguration {@link IdentityProviderConfiguration}
+     * @return User
+     * @throws IOException
+     * @throws JSONException
+     * @throws DotDataException
+     */
+    public User getUser(final HttpServletRequest request,
+                        final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        User systemUser = null;
+        User user       = null;
+
+        final HttpSession session = request.getSession(false);
+        if (session == null || request.isRequestedSessionIdValid()) {
+
+            Logger.error(this, "Could not retrieve user from session as the session doesn't exist!");
+            return null;
+        }
+
+        final String samlUserIdAttribute = identityProviderConfiguration.getId() + SAML_USER_ID;
+        if (null == session.getAttribute(samlUserIdAttribute)) {
+            return null;
+        }
+
+        final String samlUserId = (String) session.getAttribute(samlUserIdAttribute);
+        session.removeAttribute(identityProviderConfiguration.getId() + SAML_USER_ID);
+
+        try { // todo: double check if this should switch between email or user id, or this way is ok.
+            systemUser = this.userAPI.getSystemUser();
+            user       = this.userAPI.loadUserById(samlUserId, systemUser, false);
+        } catch (NoSuchUserException e) {
+
+            Logger.error(this, "No user matches ID '" + samlUserId + "'. Creating one...");
+            user = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)